vision/deskew/within_bounds.py

```diff
@@ -0,0 +1,243 @@
+    point2: [float]
+        A point that may or may not lie on the line segment that is
+        created by point1 and point3.
+        An Array that contains the x and y coordinates of the point.
+    point3: [float]
+        The end point on the line segment.
+        An Array that contains the x and y coordinates of the point.
+
+    Returns
+    -------
+    to_return: bool
+        Returns True if point2 lies on the line segment created by
+        point1 and point2. Returns False otherwise.
+
+    References
+    ----------
+    https://www.geeksforgeeks.org/check-if-two-given-line-segments-intersect/
+    """
+
+    to_return: bool = False
+
+    if ((point2[0] <= max(point1[0], point3[0])) and (point2[0] >= min(point1[0], point3[0]))) and (
+        (point2[1] <= max(point1[1], point3[1])) and (point2[1] >= min(point1[1], point3[1]))
+    ):
+        to_return = True
+
+    return to_return
+
+
+def find_orientation(point1: list[float], point2: list[float], point3: list[float]) -> int:
+    """
+    Finds the orientation of the three points. Returns int representing
+    the orientation.
+
+    Parameters
+    ----------
+    point1: [float]
+        The first point of a shape or possibly line.
+        An Array that contains the x and y coordinates of the point.
+    point2: [float]
+        The second point of a shape or possibly line.
+        An Array that contains the x and y coordinates of the point.
+    point3: [float]
+        The third point of a shape or possibly line.
+        An Array that contains the x and y coordinates of the point.
+
+    Returns
+    -------
+    to_return: int
+        Returns an integer value which represents the orientation of
+        point1, point2, and point3.
+        0 -> Colinear
+        1 -> Clockwise
+        2 -> Counterclockwise
+
+    References
+    ----------
+    https://www.geeksforgeeks.org/check-if-two-given-line-segments-intersect/
+    """
+
+    to_return: int = 0  # Defaults to_return as Colinear
+
+    # GeeksforGeeks algorithm
+    val: int = ((point2[1] - point1[1]) * (point3[0] - point2[0])) - (
+        (point2[0] - point1[0]) * (point3[1] - point2[1])
+    )
+
+    if val > 0:
+        to_return = 1  # Clockwise
+    elif val < 0:
+        to_return = 2  # Counterclockwise
+    # If val == 0, then to_return = 0
+
+    return to_return
+
+
+def lines_intersect(line1: list[list[float]], line2: list[list[float]]) -> bool:
+    """
+    Returns True if line1 and line2 intersect and False otherwise.
+
+    Parameters
+    ----------
+    line1: [[float]]
+        The first line that may or may not intersect with line2.
+        A 2D array that contains two points and each point is made up of
+        x and y coordinates.
+    line2: [[float]]
+        The second line that may or may not intersect with line1.
+        A 2D array that contains two points and each point is made up of
+        x and y coordinates.
```

Comment:
The nested data structure here would have a similar annotation in the docstring as the function definition:
`line1: [list[list[float]]]`